### ISSUE TITLE
[JDK-8222788] javafx.web build fails on XCode 10.2

### ISSUE THREAD

**Issue by johanvos** [javafxports/openjdk-jfx#439]
Since macOS 10.2, bad_optional_access is included indirectly by LLIntOffsetsExtractor.cpp from both wtf/Optional.h and the system definitions in /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/optional

```
[ 64%] Building CXX object Source/JavaScriptCore/CMakeFiles/LLIntOffsetsExtractor.dir/llint/LLIntOffsetsExtractor.cpp.o
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/src/main/native/Source/JavaScriptCore/llint/LLIntOffsetsExtractor.cpp:28:
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/src/main/native/Source/JavaScriptCore/bytecode/ArrayProfile.h:29:
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/src/main/native/Source/JavaScriptCore/runtime/Structure.h:28:
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/src/main/native/Source/JavaScriptCore/runtime/ClassInfo.h:25:
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/src/main/native/Source/JavaScriptCore/interpreter/CallFrame.h:30:
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/src/main/native/Source/JavaScriptCore/runtime/VM.h:40:
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/src/main/native/Source/JavaScriptCore/runtime/FunctionHasExecutedCache.h:29:
In file included from /Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/build/mac/Release/DerivedSources/ForwardingHeaders/wtf/StdUnorderedMap.h:28:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/unordered_map:385:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__node_handle:16:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/optional:171:70: error: redefinition of 'bad_optional_access'
class _LIBCPP_EXCEPTION_ABI _LIBCPP_AVAILABILITY_BAD_OPTIONAL_ACCESS bad_optional_access
                                                                     ^
/Users/johan/open-jfx/openjdk-jfx/modules/javafx.web/build/mac/Release/DerivedSources/ForwardingHeaders/wtf/Optional.h:281:7: note: previous definition is here
class bad_optional_access : public std::logic_error {
      ^
1 error generated.

```

Removing the definition in wtf/Optional.h fixes the problem on 10.14.4

**Comment by kevinrushforth** [comment]
@ararunprasad or @johanvos - can you file a JBS bug to track this?

**Comment by johanvos** [comment]
JBS bug: https://bugs.openjdk.java.net/browse/JDK-8222788

**Comment by ararunprasad** [comment]
It is been already addressed in https://trac.webkit.org/changeset/235661.

We just need to cherry-pick the above changeset from upstream WebKit repo.

**Comment by kevinrushforth** [comment]
This is already fixed by [JDK-8222788](https://bugs.openjdk.java.net/browse/JDK-8222788), PR #474
